### PR TITLE
Charlesmchen/webrtc/video3

### PR DESCRIPTION
**Pull request by charlesmchen**
This PR builds on #1645.

PTAL @michaelkirk

### CODE REVIEW

File: Signal/src/call/CallService.swift [hunk 1]
```diff
@@ -834,6 +843,8 @@ protocol CallServiceObserver: class {
         } else if message.hasVideoStreamingStatus() {
             Logger.debug("\(TAG) remote participant sent VideoStreamingStatus via data channel")
 
+            self.isRemoteVideoEnabled = message.videoStreamingStatus.enabled()
+
             // TODO: translate from java
```
Comment (michaelkirk) [inline Signal/src/call/CallService.swift]: TO-DONE!

File: Signal/src/call/PeerConnectionClient.swift [hunk 2]
```diff
@@ -433,7 +432,9 @@ class PeerConnectionClient: NSObject, RTCPeerConnectionDelegate, RTCDataChannelD
         Logger.debug("\(TAG) didOpen dataChannel:\(dataChannel)")
         CallService.signalingQueue.async {
             Logger.debug("\(self.TAG) set dataChannel")
+            assert(self.dataChannel == nil)
             self.dataChannel = dataChannel
+            dataChannel.delegate = self
```
Comment (michaelkirk) [inline Signal/src/call/PeerConnectionClient.swift]: 😞 . Good catch!

File: Signal/src/call/CallService.swift [hunk 3]
```diff
@@ -1026,9 +1038,26 @@ protocol CallServiceObserver: class {
         // It's only safe to access the class properties on the signaling queue, so
         // we dispatch there...
         CallService.signalingQueue.async {
-            Logger.info("\(self.TAG) \(#function): \(self.shouldHaveLocalVideoTrack())")
+            guard let call = self.call else {
+                return
```
Comment (michaelkirk) [inline Signal/src/call/CallService.swift]: I'd say both of these guards deserve a log.

Comment (charlesmchen) [inline Signal/src/call/CallService.swift]: It's expected for either `call` or `peerConnectionClient` to be nil at this point - for example, this method is called when call is set to nil.

File: Signal/src/call/CallService.swift (hunk 1 above)
Comment (charlesmchen) [inline Signal/src/call/CallService.swift]: Does that mean I should remove this commented-out code?

Comment (michaelkirk) [inline Signal/src/call/CallService.swift]: Yeah, sorry for not being clear.

**Pull request merged**